packages/deployments/contracts/contracts_forge/Connext.t.sol

```diff
@@ -90,4 +151,156 @@ contract ConnextTest is ForgeHelper {
     emit MaxRoutersPerTransferUpdated(10, address(this));
     connext.setMaxRoutersPerTransfer(10);
   }
+
+  // ============ xCall ============
+
+  // Update relayerFees mapping
+  function testXCallIncreasesRelayerFees() public {
+    address to = address(100);
+    uint256 amount = 1 ether;
+    uint256 relayerFee = 0.01 ether;
+    address transactingAssetId = address(originAdopted);
+
+    IConnext.CallParams memory callParams = IConnext.CallParams(to, bytes("0x"), domain, destinationDomain);
+    IConnext.XCallArgs memory args = IConnext.XCallArgs(callParams, transactingAssetId, amount, relayerFee);
+
+    bytes32 id = keccak256(abi.encode(0, address(this), callParams));
+
+    assertEq(connext.relayerFees(id), 0);
+
+    connext.xcall{value: relayerFee}(args);
+
+    assertEq(connext.relayerFees(id), relayerFee);
+  }
+
+  // Emit relayerFees in XCalled event
+  function testXCallEmitsRelayerFee() public {
+    address to = address(100);
+    uint256 amount = 1 ether;
+    uint256 relayerFee = 0.01 ether;
+    address transactingAssetId = address(originAdopted);
+
+    IConnext.CallParams memory callParams = IConnext.CallParams(to, bytes("0x"), domain, destinationDomain);
+    IConnext.XCallArgs memory args = IConnext.XCallArgs(callParams, transactingAssetId, amount, relayerFee);
+
+    vm.expectEmit(false, false, false, true);
+    emit XCalled(
+      bytes32("0x"),
+      to,
+      callParams,
+      address(originAdopted),
+      address(originAdopted),
+      0,
+      0,
+      relayerFee,
+      0,
+      address(this)
+    );
+    connext.xcall{value: relayerFee}(args);
+  }
+
+  // Work if relayerFee is set to 0
```

Comment:
we should probably enforce that relayer fee is always greater than 0 while the `amount` can be 0. you will always need to pay someone to relay the tx, but the tx may not include any value (i.e. updating a twap on a destination chain)